cms/djangoapps/contentstore/xblock_storage_handlers/view_handlers.py

```diff
@@ -522,6 +528,59 @@ def create_item(request):
     return _create_block(request)
 
 
+def sync_library_content(downstream: XBlock, request, store) -> StaticFileNotices:
+    """
+    Handle syncing library content for given xblock depending on its upstream type.
+    It can sync unit containers and lower level xblocks.
+    """
+    link = UpstreamLink.get_for_block(downstream)
+    upstream_key = link.upstream_key
+    if isinstance(upstream_key, LibraryUsageLocatorV2):
+        lib_block = sync_from_upstream_block(downstream=downstream, user=request.user)
+        static_file_notices = import_static_assets_for_library_sync(downstream, lib_block, request)
+        store.update_item(downstream, request.user.id)
+    else:
+        with store.bulk_operations(downstream.usage_key.context_key):
+            upstream_children = sync_from_upstream_container(downstream=downstream, user=request.user)
+            downstream_children = downstream.get_children()
+            downstream_children_keys = [child.upstream for child in downstream_children]
+            # Sync the children:
+            notices = []
+            # Store final children keys to update order of components in unit
+            children = []
+            for i in range(len(upstream_children)):
```

Comment:
Nit for later: we can use [enumerate()](https://docs.python.org/3/library/functions.html#enumerate) here.